sdk/provisioning/Azure.Provisioning.CloudMachine/src/CloudMachineInfrastructure.cs

```diff
@@ -0,0 +1,216 @@
+            RequiresSession = false,
+            DefaultMessageTimeToLive = new StringLiteral("P14D"),
+            DeadLetteringOnFilterEvaluationExceptions = true,
+            DeadLetteringOnMessageExpiration = true,
+            MaxDeliveryCount = 10,
+            EnableBatchedOperations = true,
+            Status = ServiceBusMessagingEntityStatus.Active
+        };
+        _eventGridTopic_Blobs = new($"{_name}_eg_blob", "2022-06-15")
+        {
+            TopicType = "Microsoft.Storage.StorageAccounts",
+            Source = _storage.Id,
+            Identity = managedServiceIdentity
+        };
+        _systemTopicEventSubscription = new($"{_name}_eg_blob_sub", "2022-06-15")
+        {
+            Parent = _eventGridTopic_Blobs,
+            DeliveryWithResourceIdentity = new DeliveryWithResourceIdentity
+            {
+                Identity = new EventSubscriptionIdentity
+                {
+                    IdentityType = EventSubscriptionIdentityType.UserAssigned,
+                    UserAssignedIdentity = _identity.Id
+                },
+                Destination = new EventHubEventSubscriptionDestination
+                {
+                    ResourceId = _serviceBusTopic_main.Id
+                }
+            },
+            Filter = new EventSubscriptionFilter
+            {
+                IncludedEventTypes =
+                [
+                    "Microsoft.Storage.BlobCreated",
+                    "Microsoft.Storage.BlobDeleted",
+                    "Microsoft.Storage.BlobRenamed"
+                ],
+                IsAdvancedFilteringOnArraysEnabled = true
+            },
+            EventDeliverySchema = EventDeliverySchema.EventGridSchema,
+            RetryPolicy = new EventSubscriptionRetryPolicy
+            {
+                MaxDeliveryAttempts = 30,
+                EventTimeToLiveInMinutes = 1440
+            }
+        };
+    }
+
+    public override ProvisioningPlan Build(ProvisioningContext? context = null)
+    {
+        // Always add a default location parameter.
+        // azd assumes there will be a location parameter for every module.
+        // The Infrastructure location resolver will resolve unset Location properties to this parameter.
+        Add(new BicepParameter("location", typeof(string))
+        {
+            Description = "The location for the resource(s) to be deployed.",
+            Value = BicepFunction.GetResourceGroup().Location
+        });
+
+        Add(PrincipalIdParameter);
+        Add(PrincipalTypeParameter);
+        Add(PrincipalNameParameter);
+
+        Add(_identity);
+        Add(_storage);
+        Add(_storage.AssignRole(StorageBuiltInRole.StorageBlobDataContributor, RoleManagementPrincipalType.User, PrincipalIdParameter));
+        Add(_storage.AssignRole(StorageBuiltInRole.StorageTableDataContributor, RoleManagementPrincipalType.User, PrincipalIdParameter));
+        Add(_container);
+        Add(_blobs);
```

Comment:
It might be better to parent this to the storage account here to prepare for when things open up.

Reply:
The suggestion is that you define parenting here at "compose time," like you're defining roles here, so when you make things customizable someone can just replace the `BlobService` without having to properly link everything.